ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh

```diff
@@ -40,7 +40,48 @@ bmc_password = sys.argv[3]
 iso_path = sys.argv[4]
 transfer_protocol_type = sys.argv[5]
 
-context = redfish.redfish_client(bmc_address, username=bmc_username, password=bmc_password)
+retry_counter = 0
+
+def redfish_mount_remote(context):
+    global retry_counter
+    response = context.post(f"/redfish/v1/Managers/{manager}/VirtualMedia/{removable_disk}/Actions/VirtualMedia.InsertMedia",
+                        body={"Image": iso_path, "TransferProtocolType": transfer_protocol_type,
+                              "Inserted": True, **other_options})
+    print(f"/redfish/v1/Managers/{manager}/VirtualMedia/{removable_disk}/Actions/VirtualMedia.InsertMedia")
+    print({"Image": iso_path, "TransferProtocolType": transfer_protocol_type})
+
+    #print(response.__dict__)
+    print(response.status)
+    print(response.text)
+    if response.status > 299:
+        sys.exit(1)
+    d = {}
+    task = None
+    if response.is_processing:
+        while task is None or (task is not None and
+                            (task.is_processing or not task.dict.get("TaskState") in ("Completed", "Exception"))):
+            task = response.monitor(context)
+            print("Task target: %s" % bmc_address)
+            print("Task is_processing: %s" % task.is_processing)
+            print("Task state: %s " % task.dict.get("TaskState"))
+            print("Task status: %s" % task.status)
+            #print("Task dict: %s" % task.dict)
+            retry_time = task.retry_after
+            time.sleep(retry_time if retry_time else 5)
+            if (task.dict.get("TaskState") in ("Exception")):
+               if (retry_counter < 6):
+                    print("Exception in %s task Mount Remote, retrying" % bmc_address)
+                    retry_counter=retry_counter+1
+                    redfish_mount_remote(context=context)
```

Comment:
hi @bmanzari, thanks for the PR. If we could avoid the recursion by using a for loop, I'd prefer it. If you still prefer to use recursion, I'd let the `retry_counter` var be a local parameter of the function.

But in general, it looks good to me. Once you conclude testing and remove the WIP, I'm ok to approve it.

Reply:
now using a loop and local vars, let me know